src/Nri/Ui/Checkbox/V7.elm

```diff
@@ -0,0 +1,494 @@
+
+-}
+labelCss : List Css.Style -> Attribute msg
+labelCss styles =
+    Attribute <| \config -> { config | labelCss = config.labelCss ++ styles }
+
+
+{-| Hides the visible label. (There will still be an invisible label for screen readers.)
+-}
+hiddenLabel : Attribute msg
+hiddenLabel =
+    Attribute <| \config -> { config | hideLabel = True }
+
+
+{-| Shows the visible label. This is the default behavior
+-}
+visibleLabel : Attribute msg
+visibleLabel =
+    Attribute <| \config -> { config | hideLabel = False }
+
+
+{-| Set a custom ID for this checkbox input and label. If you don't set this,
+we'll automatically generate one from the label you pass in, but this can
+cause problems if you have more than one checkbox input with the same label on
+the page. You might also use this helper if you're manually managing focus.
+-}
+id : String -> Attribute msg
+id id_ =
+    Attribute <| \config -> { config | id = Just id_ }
+
+
+{-| Use this helper to add custom attributes.
+
+Do NOT use this helper to add css styles, as they may not be applied the way
+you want/expect if underlying styles change.
+Instead, please use the `css` helper.
+
+-}
+custom : List (Html.Attribute Never) -> Attribute msg
+custom attributes =
+    Attribute <| \config -> { config | custom = config.custom ++ attributes }
+
+
+{-| -}
+nriDescription : String -> Attribute msg
+nriDescription description =
+    custom [ Extra.nriDescription description ]
+
+
+{-| -}
+testId : String -> Attribute msg
+testId id_ =
+    custom [ Extra.testId id_ ]
+
+
+{-| Customizations for the Checkbox.
+-}
+type Attribute msg
+    = Attribute (Config msg -> Config msg)
+
+
+{-| This is private. The public API only exposes `Attribute`.
+-}
+type alias Config msg =
+    { id : Maybe String
+    , hideLabel : Bool
+    , onCheck : Maybe (Bool -> msg)
+    , isDisabled : Bool
+    , guidance : Guidance
+    , custom : List (Html.Attribute Never)
+    , containerCss : List Css.Style
+    , labelCss : List Css.Style
+    }
+
+
+{-|
+
+    = Selected --  Checked (rendered with a checkmark)
+    | NotSelected -- Not Checked (rendered blank)
+    | PartiallySelected -- Indeterminate (rendered dash)
+
+-}
+type IsSelected
+    = Selected
+    | NotSelected
+    | PartiallySelected
+
+
+emptyConfig : Config msg
+emptyConfig =
+    { id = Nothing
+    , hideLabel = False
+    , onCheck = Nothing
+    , isDisabled = False
+    , guidance = InputErrorAndGuidanceInternal.noGuidance
+    , custom = []
+    , containerCss = []
+    , labelCss = []
+    }
+
+
+applyConfig : List (Attribute msg) -> Config msg -> Config msg
+applyConfig attributes beginningConfig =
+    List.foldl (\(Attribute update) config -> update config)
+        beginningConfig
+        attributes
+
+
+{-| View a checkbox (the label is only used for accessibility hints unless visibleLabel attribute is applied)
+-}
+view :
+    { label : String
+    , selected : IsSelected
```

Comment:
Why don't we add onCheck to the config as well? It's never not set.
Additionally, I wonder if we should force adding an id by moving it to the config.

Reply:
I would not mind moving them, but so far all components have the event handler and id setter in the attributes list. The mandatory arguments are the requirements for using them in regular forms + a11y if I understand correctly the design.

Reply:
The concept for having onCheck/other event handlers not in the main record is that as a dev you can make progress _towards_ using a component incrementally. Instead of needing to understand a big record full of options, I can start with something like: `Button.button "My button" []` and have _something_ show up, and I can work through everything else I want one piece at a time.

Also it makes the "preview" and example code features easier to support